spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/StreamCompletionProvider.java

```diff
@@ -35,6 +39,15 @@
 	@Autowired
 	private List<? extends ExpansionStrategy> completionExpansionStrategies = new ArrayList<>();
 
+	@PostConstruct
+	public void sortStrategies() {
+		// Here we mean HIGHEST_PRECEDENCE to win over others, so must be *last*
+		AnnotationAwareOrderComparator.sort(completionExpansionStrategies);
```

Comment:
I'm not sure I understand how field injection is the culprit here. While I agree with the move to constructor injection, what's wrong with `@Ordered`? Are you saying `StreamCompletionProvider` should know about the desired order?